src/libexpr/primops.cc

```diff
@@ -1638,6 +1638,29 @@ static RegisterPrimOp primop_hashFile({
     .fun = prim_hashFile,
 });
 
+
+static void prim_readFileType(EvalState & state, const PosIdx pos, Value * * args, Value & v)
```

Comment:
```suggestion
static void prim_pathType(EvalState & state, const PosIdx pos, Value * * args, Value & v)
```

Reading is an implementation detail, and it's not necessarily a file.
See https://github.com/NixOS/nix/issues/3096
Also I think it would be nice to let the caller handle the missing file case themselves
> `null` when the path p does not point to an existing filesystem node

Otherwise, some functions will need to do `pathExists` first, which amounts to another `stat`.